src/loki/search.cc

```diff
@@ -303,9 +303,11 @@ struct bin_handler_t {
                                            PathLocation::NONE,
                                            reach.outbound,
                                            reach.inbound};
-          auto index = other_edge->forward() ? 0 : info.shape().size() - 2;
-          if (heading_filter(other_edge, tile->edgeinfo(edge->edgeinfo_offset()), location,
-                             candidate.point, index)) {
+          // index is opposite the logic above, plus we need to use the edgeinfo and shape
+          // from the other tile since we use the other_edge forward flag
+          auto index = other_edge->forward() ? info.shape().size() - 2 : 0;
+          if (heading_filter(other_edge, other_tile->edgeinfo(other_edge->edgeinfo_offset()),
```

Comment:
the only time the other edge info could be different is if they had different names, it technically isnt a bug to use the edge info from not this edge as in terms of the shape and direction of the shape, there will be no difference.

only reason im saying this is because i purposely didnt use the opposing edges shape because the decoded shape gets cached on access, so in the case that it is a differetn tile with a different edge info (with the same shape), then we needlessly decode it again. i get that this is more clear, but it cant be less performant.

probably a comment mentioning why its ok to use the same shape from the other edge would be sufficient? or we can ignore the potential performance ramifications. what are your thoughts?

Reply:
also, thanks for catching the index thing!

Reply:
@dnesbitt61 the same bug exists on line 380 with regard to the index. we'll need to amend the pr to subtract the candidate index from the total shape size in that case right?

Reply:
problem with using the shape from the original edge is that if the opposing edge is in a different tile it will have its own edge info and shape with the forward flag set to true. So I'm pretty sure there can be an incompatibility here since both directed edges will have forward == true. Would need to bypass the forward flag (e.g., send it in as an argument rather than using the directed edge).

Reply:
also note that you could compute the angle once and then use the angle 180 degrees for the opposing edge...trick is how to do this when other edge filters are in play...